### DISCUSSION TITLE
Plot Package

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#432] [General]
Hello all,

I recently implemented some GUI data plot (i.e. MeshSolution data, still internal but to come) and had some (unsolved) issues with pyvista(qt). This let me to some thoughts about ploting data in Pyleecan/SciDataTool in general. I.e. with matplotlib embedded in the GUI there seems to be limited capabilites on the figures (controls and data acquisition).
Also recently I found [pyqtgraph](https://www.pyqtgraph.org/) package which seems to be a good alternative to matplotlib. If one don't want to install it, [this youtube](https://youtu.be/yhU-HUTJ8Y0) may give some good overview of the examples.

So my actual question is, is there anyone who knows both of these packages and can give some pros and cons if we would like to switch from matplotlib to pyqtgraph for data vis.?

Best regards, Sebastian

**Comment by BonneelP** [comment]
Hello Sebastian,

This package looks interesting but I have never used it sorry :/

I would say that as pyleecan can offer several models for the same purpose we can have several options to plot the same things. The plan for the SciDataTool plot GUI is to spilt the "get the correct data on the proper axis" and the "plot it" parts. Basically a widget to interact with get_along then a "simple" call to plot_2D (according to what was returned). So we can imagine to have two different widget to interact with the "plot it" part, one for matplotlib and one for pyqtgraph. Hope it help you in your thoughts. 

Best regards,
Pierre

**Comment by EmileDvs** [comment]
Hello @SebGue,

I am very glad to read that you are currently making an interface to interact with Meshsolution objects within pyleecan and I am looking forward to testing it!

As @BonneelP said, we are indeed currently working on an interface to plot SciDataTool objects and the next step would be the same for Meshsolution objects. 

Concerning Meshsolution plots, I have been wondering for a while how to improve pyleecan postprocessings to get flux lines and smoother field interpolation (such as FEMM interface provides), since at the moment Meshsolution plot yields constant value per element without the possibility to draw flux lines. Maybe pyvista also provides methods to do so but I haven't got much time to investigate.  

Have you maybe thought about this on your side ? I ask this because it seems to me that it is quite linked to the other discussion you recently started (#441), which deals with calculating some quantities from the others.

Best regards,
Emile

**Comment by SebGue** [comment]
Hello @EmileDvs 

thats what I got so far (see figure). Its some simple results visualisation in matplotlib, but I can push it right now.
![grafik](https://user-images.githubusercontent.com/43440731/134626669-b1a519f2-0df8-4141-86ab-6be11f7777f9.png)

Is there already a branch for the SciDataTool plot interface?

For the flux lines we need the mag. vector potential as you know. So it's indeed linked to the other discussion. By the way, what is  @Superomeg4 opinion here?

I read that the FEMM interpolation is just some 'trick' to have a nicer view, but I don't know how its done. I think it was based on the field values and the triangles gauss points. But maybe this can also been done with an interpolation of Az. At least there are some methods to do scattered data interpolation in scipy package. 

Best regards, Sebastian
